### ISSUE TITLE
When using @ValueMapping string-to-enum mapping results in no cases in switch

### ISSUE THREAD

**Issue by jcbodnar** [mapstruct/mapstruct#2350]
My code:

```
@Mapper
public interface TransactionMapper {
    @Mapping(source = "fullName", target = "name")
    @Mapping(expression = "java(payload.getDonorFirstName() + \" \" + payload.getDonorLastName())", target = "donorName")
    @Mapping(source = "donorEmailAddress", target = "donorEmail")
    @Mapping(constant = "false", target = "processed")
    @Mapping(source = "schedule.givingFrequency", target = "recurringFrequency")
    @Mapping(source = "schedule.startDate", target = "recurringStartDate")
    @Mapping(source = "schedule.processImmediately", target = "processImmediately")
    TransactionEntity payloadToEntity(TransactionsPayload payload);

    @ValueMapping(source = "ONETIME", target = "ONE_TIME")
    TransactionType transactionTypeStringToTransactionType(String transactionTypeString);
}
```

The possible `String` values for `transactionTypeString` are `ONETIME`, `RECURRING` and `PLEDGE`. The `TransactionType` enum has values of `ONE_TIME`, `RECURRING` and `PLEDGE`. Not the underscore in `ONE_TIME`. That's why I have to specify the mapping.

The code generated has no `case` statements for each value and only the `default` that throws an exception:

```
@Override
    public TransactionType transactionTypeStringToTransactionType(String transactionTypeString) {
        if ( transactionTypeString == null ) {
            return null;
        }

        TransactionType transactionType;

        switch ( transactionTypeString ) {
            default: throw new IllegalArgumentException( "Unexpected enum constant: " + transactionTypeString );
        }
    }
```

If I change my mapper to:

```
@ValueMapping(source = "ONETIME", target = "ONE_TIME")
@ValueMapping( source = MappingConstants.ANY_REMAINING, target = "RECURRING" )
TransactionType transactionTypeStringToTransactionType(String transactionTypeString);
```

the generated code is:

```
@Override
    public TransactionType transactionTypeStringToTransactionType(String transactionTypeString) {
        if ( transactionTypeString == null ) {
            return null;
        }

        TransactionType transactionType;

        switch ( transactionTypeString ) {
            case "ONETIME": transactionType = TransactionType.ONE_TIME;
            break;
            case "ONE_TIME": transactionType = TransactionType.ONE_TIME;
            break;
            case "RECURRING": transactionType = TransactionType.RECURRING;
            break;
            case "PLEDGE": transactionType = TransactionType.PLEDGE;
            break;
            default: transactionType = TransactionType.RECURRING;
        }

        return transactionType;
    }
```

which is close. I need the `default` to still throw an exception.

**Comment by filiphr** [comment]
Thanks for raising this issue @jcbodnar. It should be fixed with PR #2354. 

I think that this slipped up when we were implementing this feature, initially we didn't allow having a String to Enum mapping without any remaining or any unmapped, but then we allowed it, but forgot to update some paths in our code :)

**Comment by Delirante** [comment]
Works in 1.5.0.Beta2.
